### PR TITLE
Plumb through tensor.unpack e2e execution for llvm-cpu backend.

### PR DESCRIPTION
**Pull request by hanhanW**
This has the same test coverage as iree_linalg_ext.unpack version.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 1]
```diff
@@ -1095,14 +1095,21 @@ static LogicalResult setPackOpRootConfig(func::FuncOp entryPointFn, OpTy op) {
...
+template <typename OpTy>
+static LogicalResult setUnPackOpRootConfig(
+    func::FuncOp entryPointFn, OpTy op,
     DispatchLoweringPassPipeline pipeline =
         DispatchLoweringPassPipeline::CPUDataTiling) {
+  // TODO(hanchung): Retire IREE::LinalgExt::UnPackOp. This is for having
+  // consistent configurations for unpack ops.
+  static_assert(
+      llvm::is_one_of<OpTy, IREE::LinalgExt::UnPackOp, tensor::UnPackOp>::value,
+      "applies to only pack operations");
```
Comment (bjacob) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: `unpack` ?

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 2]
```diff
@@ -1095,14 +1095,21 @@ static LogicalResult setPackOpRootConfig(func::FuncOp entryPointFn, OpTy op) {
...
+
   // TODO(#11505): Consider multi-level tiling for handling unpack + generic
   // cases.
-  SmallVector<int64_t> tileSizes =
-      getLinalgExtDefaultWorkgroupTileSizes(op, /*defaultSize=*/16);
+  SmallVector<int64_t> tileSizes = getLinalgExtDefaultWorkgroupTileSizes(
+      cast<TilingInterface>(op.getOperation()), /*defaultSize=*/16);
```
Comment (bjacob) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: Rebasing this past #12134 means changing the `16` here to `defaultWorkgroupTileSizeForUnpackOp`.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp [hunk 3]
```diff
@@ -678,8 +678,10 @@ static unsigned decideFusableLinalgOps(FunctionOpInterface funcOp,
       // Only look for Linalg ops here. Avoid moving `linalg.fill` that aren't
```
Comment (bjacob) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp]: Update this comment?

**Review by bjacob — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp (hunk 1 above)
Comment (bjacob) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: Though my favorite choice is no message at all (if we can assume C++17) or empty string `""` (before C++17) as the assert condition here `llvm::is_one_of<OpTy, IREE::LinalgExt::UnPackOp, tensor::UnPackOp>::value` is easy to understand and the message does not add more useful information (and, in this case, was potentially inaccurate).

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp (hunk 1 above)
Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: yes, I like it better as well! removed the message.

**Pull request merged**